web/components/contract/contract-info-dialog.tsx

```diff
@@ -147,3 +176,69 @@ const getTweetText = (contract: Contract, isCreator: boolean) => {
 
   return `${tweetQuestion}\n\n${tweetDescription}\n\n${url}`
 }
+
+export function EditableResolutionTime(props: {
+  time: number
+  contract: Contract
+  isCreator: boolean
+}) {
+  const { time, contract, isCreator } = props
+
+  const [isEditing, setIsEditing] = useState(false)
+  const [timeString, setTimeString] = useState(time && formatTime(time))
```

Comment:
I don't really think that the UI where you click the pencil and it becomes a writable input has any advantages at all over the UI where it's just an input in the first place, and it has disadvantages (like complexity, probable inaccessibility, and you have to click the pencil.)

Reply:
I agree in general, but I don't like the way the datepicker doesn't fit in with the other fields:
![image](https://user-images.githubusercontent.com/61841994/172500059-c80b6584-24ed-42f7-82fd-c0185c05469e.png)

It's probably good to think about it some more outside of this feature. E.g. there is no reason for me the close date shouldn't be editable there as well.

Reply:
Agree their's room for improvement here, in terms of how you edit the dates and also allowing the market close to be edited here, which is fairly important.
I'm ok for that to be a followup PR.